deepchem/data/data_loader.py

```diff
@@ -1408,30 +1448,94 @@ def create_dataset(self,
             input_files = remainder
 
         # Sort image files
-        image_files = sorted(image_files)
+        if self.sorting:
+            image_files = sorted(image_files)
+
+        if isinstance(labels, str):
+            label_files = [labels]
+        else:
+            label_files = []
+
+        label_image_files = []
+        # Sometimes zip files contain directories within. Traverse directories
+        while len(label_files) > 0:
+            remainder = []
+            for label_file in label_files:
+                filename, extension = os.path.splitext(label_file)
+                extension = extension.lower()
+                if os.path.isdir(label_file):
+                    dirfiles = [
+                        os.path.join(label_file, subfile)
+                        for subfile in os.listdir(label_file)
+                    ]
+                    remainder += dirfiles
+                elif extension == ".zip":
```

Comment:
@aaronrockmenezes Do zip files have a set ordering upon extraction? The example above uses a label zip and a data directory zip. Do we know ordering is preserved or is it implementation dependent and can change?

Reply:
All files within a zipped folder are extracted in the order they appear in the directory that is to be extracted. So yes, as far as I know, the order is maintained.